pandas/core/frame.py

```diff
@@ -2590,7 +2590,7 @@ def assign(self, **kwargs):
 
         return data
 
-    def _sanitize_column(self, key, value):
+    def _sanitize_column(self, key, value, broadcast=False):
```

Comment:
I think it would be better to do the opposite here -- keep the default behavior of `_sanitize_column` the same and override in the case of `insert` where you specify the column. The `insert` case where you don't want to broadcast the value is the special one. The more common case is: 
```
df1 = pd.DataFrame({'a': [1, 2, 3]})
df2 = pd.DataFrame({'b': [2,3,4]})
df = pd.concat([df1, df1, df2], axis=1)
df.index = [1, 2, 3]
df.loc[1, 'a'] = 3
```
You would override the defaults less often if you kept the default as `True` and only overrode it in the special case of `insert`.